test/tests/functional/pbs_mom_job_create.py

```diff
@@ -38,34 +38,71 @@
 from tests.functional import *
 
 
-class TestMomJobCreate(TestFunctional):
+class TestMomJobDir(TestFunctional):
     """
     This test suite tests the mom's ability to create job directories.
     """
 
-    def test_existing_directory(self):
-        """
-        If a job directory already exists, the mom should clean it up
-        """
+    def change_server_name(self, servername):
         self.server.stop()
         self.assertFalse(self.server.isUp(), 'Failed to stop PBS')
 
-        servername = 'superlongservername'
         conf = self.du.parse_pbs_config(self.server.hostname)
         self.du.set_pbs_config(
-            self.server.shortname,
+            self.server.hostname,
             confs={'PBS_SERVER_HOST_NAME': conf['PBS_SERVER'],
                    'PBS_SERVER': servername})
 
         self.server.start()
         self.assertTrue(self.server.isUp(), 'Failed to start PBS')
+        return
+
+    def test_existing_directory_longid(self):
+        """
+        If a job directory already exists, the mom should clean it up
+        after rejecting the job. When the server sends the request for
+        the second time, the mom will run the job correctly.
+        The mom has special code if the job id is less than 11 characters,
+        so the job id must be longer than 11 characters.
+        """
+
+        # Change the server name to create a job id longer than 11 characters
+        self.change_server_name('superlongservername')
 
         j = Job(TEST_USER, attrs={ATTR_h: None})
         j.set_sleep_time(3)
         jid = self.server.submit(j)
+
+        conf = self.du.parse_pbs_config(self.server.hostname)
```

Comment:
@vstumpf , I have a question here . Wouldn't self.server.pbs_conf already be updated now. Do we need to parse the config again here ? Can't we grab self.server.pbs_conf directly ?

Reply:
`Server.pbs_conf` doesn't get updated when we parse configs or set configs. I could edit `change_server_name` to save conf to the test, but it's not an obvious side effect.